src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.dao.impl.BetDaoImpl;
+import core.basesyntax.dao.impl.UserDaoImpl;
+import core.basesyntax.exceptions.NoAnnotationException;
+import core.basesyntax.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException,
+            InstantiationException, NoAnnotationException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                field.setAccessible(true);
+                Class fieldType = field.getType();
```

Comment:
It's better to generify the Class variable for the case of widening our project structure

Reply:
I can use any class with this variable. If i generify Class with <? extends UserDao or BetDao> i narrow the field of view to just classes which extends UserDao or BetDao

Reply:
That's exactly what we need here. I was talking about potential classes extending UserDao or BetDao